### PR TITLE
iBOT Implementation

### PR DESCRIPTION
**Pull request by yutong-xiang-97**
Implement iBOT in benchmarks

Difference from DINOv2:

- Teacher and student heads are always shared. The only difference between teacher and student heads is `norm_last_layer`
- Head output_dim is 8192
- Final teacher temp for both the cls(DINO) loss and patch (IBOTPatch) loss is 0.07
- init lr 0.0005 with linear scaling
- No layerwise lr decay
- momentum start value is 0.996

### CODE REVIEW

File: benchmarks/imagenet/vitb16/ibot.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+            (cls_token.detach(), targets), batch_idx
+        )
+        self.log_dict(cls_log, prog_bar=True, sync_dist=True, batch_size=len(targets))
+        return cls_loss
+
+    def configure_optimizers(self):
+        # Don't use weight decay for batch norm, bias parameters, and classification
+        # head to improve performance.
+        min_lr = 1e-6
+        lr_scale = self.batch_size_per_device * self.trainer.world_size / 1024
```
Comment (stegmuel) [inline benchmarks/imagenet/vitb16/ibot.py:208]: In [iBOT ](https://github.com/bytedance/ibot/blob/da316d82636a7a7356835ef224b13d5f3ace0489/main_ibot.py#L306)they scale the lr using 256 as reference batchsie

**Review by stegmuel — CHANGES_REQUESTED** [review]
One small change, otherwise all good.

**Review by stegmuel — APPROVED** [review]
LGTM!
